libGitWrap/Private/DiffPrivate.hpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * libGitWrap - A Qt wrapper library for libgit2
+ * Copyright (C) 2015 The MacGitver-Developers <dev@macgitver.org>
+ *
+ * (C) Nils Fenner <nils@macgitver.org>
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License (Version 2) as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+
+#include "libGitWrap/Private/IndexPrivate.hpp"
+#include "libGitWrap/Private/RepoObjectPrivate.hpp"
+#include "libGitWrap/Private/TreePrivate.hpp"
+
+
+namespace Git
+{
+
+    class Diff;
+
+    namespace Internal
+    {
+
+        class DiffOptions : public QSharedData
+        {
+        public:
+            typedef QExplicitlySharedDataPointer<DiffOptions> Ptr;
+
+        public:
+            DiffOptions( Result& result )
+                : mOpts( new git_diff_options )
+            {
+                initOpts( result );
+            }
+
+            DiffOptions(Result& result, git_diff_options* opts, bool init = false)
+                : mOpts( opts )
+            {
+                if ( init ) {
+                    initOpts( result );
+                }
+            }
+
+            ~DiffOptions()
+            {
+                // TODO: clear mOpts
+                delete mOpts;
+            }
+
+        private:
+            void initOpts( Result& result )
+            {
+                Q_ASSERT( mOpts );
+                result = git_diff_init_options( mOpts, GIT_DIFF_OPTIONS_VERSION );
+            }
+
+        public:
+            git_diff_options*   mOpts;
+        };
+
+
+        class DiffListPrivate : public RepoObjectPrivate
+        {
+        public:
+            DiffListPrivate(const RepositoryPrivate::Ptr& repo, git_diff* diff);
+            ~DiffListPrivate();
+
+        public:
+            git_diff* mDiff;
+        };
+
+
+        class DiffFilePrivate : public BasePrivate
+        {
+        public:
+            DiffFilePrivate(const git_diff_file* diffFile);
+            ~DiffFilePrivate();
+
+        public:
+            const git_diff_file* mDiffFile;
+        };
+
+
+        class DiffBasePrivate : public BasePrivate
+        {
+        public:
+            DiffBasePrivate(Result& result)
+                : BasePrivate()
+            {
+                mOpts = new DiffOptions( result );
```

Comment:
Ahh, now I understand what you mean. This piece of code is the unfinished part. I'll correct it.